### ISSUE TITLE
[Chef 18] chef_client_config breaks with `rights` in the current code base

### ISSUE THREAD

**Issue by mvangoor** [chef/chef#14170]
The way this resource works has changed since Chef 17 where if the directory existed it just did nothing. I can see how that would not deliver desired state configuration outcome and agree this was not the best course of action.

We are keen on security and have had the following code snippet in our company baseline cookbook to ensure the chef folder is secure.
```ruby
  directory 'c:\chef' do
    only_if { File.directory?('c:\chef') }
    rights :full_control, 'Administrators'
    rights :full_control, 'SYSTEM'
    inherits false
    recursive true
  end
```

If you add this to a chef infra client 18 run you end up in a chicken/egg issue where the current resource overrides this, or tries to, as it does not use the `rights` option, which is windows only...

The current chef_client_config resource does not provide options to setup rights OR windows specific requirements as such.

My first idea was to make a distinction here between windows and rest and provide the rights option on the resource, but am not sure that is the best way forward.

The relevant code is here for infra client 18:
https://github.com/chef/chef/blob/4d66a5dc552a37f856dc59516a31d629d44a4769/lib/chef/resource/chef_client_config.rb#L256-L271

Curious to hear your thoughts around how we can achieve the lockdown we desire and not break the complete chef folder.

For the record:
doing chef_client_config and this code leaves the c:\chef folder without any permissions at all. We needed to do takeown and icacls to get access and restore permissions and it broke directly again so it is reproduceable.

**Comment by johnmccrae** [comment]
There is a weirdo bug that appears to be a Ruby problem. In the file_access_control\windows.rb file the code incorrectly detects a nil object - acls.nil? returns false when it is in fact nil. That results in a borked ACL object that fails and resets permissions or otherwise does bad things. I'm working on a PR to address that.

**Comment by johnmccrae** [comment]
Corrected with this [PR](https://github.com/chef/chef/pull/14173). Released in Chef 18.4.x
